src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -283,6 +283,12 @@ private EventHubAdapterReceiver GetOrCreateReceiver(QueueId queueId)
             return cache;
         }
 
+        protected virtual IEventHubQueueCache CreateCache(IStreamQueueCheckpointer<string> checkpointer, 
```

Comment:
We can provide a default implementation of that class, and user can extend it through inheritance or decorating. In this way, we can make EventHubAdapterFactory (a already complicated class) simpler and cleaner, and bring more flexibility in the future to CreateCacheFactory class if we want to do more customization. 

We can also do a ICreateCacheFactory interface, and provide a default implementation for it. I will leave it to you to decide pros or cons on those alternatives. 

I like what you are intending to do in this PR. But I'd like to see it be done in a less confusing way.